commons/src/main/java/org/sonarsource/analyzer/commons/PatternDirectoryScanner.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarSource Analyzers Commons
+ * Copyright (C) 2009-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.analyzer.commons;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.filefilter.IOFileFilter;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.sonar.api.utils.WildcardPattern;
+
+public class PatternDirectoryScanner {
```

Comment:
now I think if we should name it `DirectoryPatternProvider` :) seems to fit better

Reply:
You will have to convince me because it sounds weird if I try to translate that class name into a sentence: `this class is a provider of pattern directories`. To be honest, I actually liked the original name the Python bubble came up with.

Reply:
But scanner should have `scan` method with some behaviour applied to each scanned item. But I agree that my option is also not perfect :)

Reply:
I think you are a bit influenced by the semantics (and API) of scanners at SonarSource :), and now that I think about it, it does make sense not to use the term "scanner" here to avoid the confusion.

What about `FileProvider` then ?

Reply:
sounds good